### PR TITLE
F-10: Updated model according to latest changes in DB scheme

### LINKED ISSUE #10: Update model according to latest changes in DB scheme

### PR DESCRIPTION
**Pull request by venmer**
closes #10

### CODE REVIEW

File: flotski/fbooking/models.py [hunk 1]
```diff
@@ -30,24 +21,25 @@ class User(models.Model):
 class Booking(models.Model):
     start_date = models.DateField()
     end_date = models.DateField()
-    user_id = models.ForeignKey(User, on_delete=models.CASCADE)
-    room_id = models.ForeignKey(Room, on_delete=models.CASCADE)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    room = models.ForeignKey(Room, on_delete=models.CASCADE)
```
Comment (venmer) [inline flotski/fbooking/models.py]: found that Django automatically extends key fields with `_id` postfix.

File: flotski/fbooking/models.py [hunk 2]
```diff
@@ -30,24 +21,25 @@ class User(models.Model):
...
     creation_date = models.DateField(auto_now_add=True)
 
 class Guest(models.Model):
     first_name = models.CharField(max_length=35)
     last_name = models.CharField(max_length=35)
     passport = models.CharField(max_length=15, unique=True)
-    birthdate = models.DateField()
+    birth_date = models.DateField()
```
Comment (venmer) [inline flotski/fbooking/models.py]: renamed

File: flotski/fbooking/models.py [hunk 3]
```diff
@@ -4,16 +4,7 @@
...
-    VACANT = 'V'
-    OCCUPIED = 'O'
-    MAINTENANCE = 'M'
-    ROOM_STATE_CHOICES = (
-        (VACANT, 'Vacant'),
-        (OCCUPIED, 'Occupied'),
-        (MAINTENANCE, 'Maintenance')
-    )
     description = models.CharField(max_length=255, null=True)
-    state = models.CharField(max_length=1, choices=ROOM_STATE_CHOICES, default=VACANT)
```
Comment (venmer) [inline flotski/fbooking/models.py]: as was discussed - `state` field removed

File: flotski/fbooking/tests.py [hunk 4]
```diff
@@ -160,72 +160,78 @@ def test_update_room(self):
...
+        [None, "UserName", "123456", datetime.datetime(1992, 1, 23), True],
+        ["FirstName", None, "123456", datetime.datetime(1992, 1, 23), True],
+        ["FirstName", "UserName", None, datetime.datetime(1992, 1, 23), True],
+        ["FirstName", "UserName", "123456", None, True],
+        ["FirstName", "UserName", "123456", datetime.datetime(1992, 1, 23), False]
     ])
-    def test_add_guest_without_required_properties(self, first_name, last_name, passport, birthdate):
-        guest = self._prepare_test_guest(first_name, last_name, passport, birthdate)
+    def test_add_guest_without_required_properties(self, first_name, last_name, passport, birth_date, user):
+        guest = self._prepare_test_guest(first_name, last_name, passport, birth_date, self.user if user else None)
```
Comment (venmer) [inline flotski/fbooking/tests.py]: `@parameterized` gets called before test class init, so I cannot access `self` object. I think it's ok to do it this way.

**Comment by venmer** [comment]
@Bright-Head, please review.

File: flotski/fbooking/models.py (hunk 1 above)
Comment (Bright-Head) [reply flotski/fbooking/models.py]: Ok

File: flotski/fbooking/tests.py (hunk 4 above)
Comment (Bright-Head) [reply flotski/fbooking/tests.py]: Agreed

File: flotski/fbooking/models.py [hunk 5]
```diff
@@ -4,16 +4,7 @@
...
 class Room(models.Model):
-    VACANT = 'V'
-    OCCUPIED = 'O'
-    MAINTENANCE = 'M'
-    ROOM_STATE_CHOICES = (
-        (VACANT, 'Vacant'),
-        (OCCUPIED, 'Occupied'),
-        (MAINTENANCE, 'Maintenance')
-    )
     description = models.CharField(max_length=255, null=True)
```
[the remark this answers is not in the document]
Comment (venmer) [reply flotski/fbooking/models.py]: ok, but Room `description`  is also might be useful in giving some extra information like floor, view or whatever.

File: flotski/fbooking/models.py [hunk 6]
```diff
@@ -12,10 +12,18 @@ class Permission(models.Model):
     description = models.CharField(max_length=255)
 
 class User(models.Model):
+    ACTIVE = 1
+    INACTIVE = 0
+    USER_STATE = (
+        (ACTIVE, 'active'),
+        (INACTIVE, 'inactive'),
+    )
+    description = models.CharField(max_length=255, null=True)
```
[the remark this answers is not in the document]
Comment (venmer) [reply flotski/fbooking/models.py]: yes, looks like a copy-paste.
and it seems that Django doesn't handle duplicated fields:
`
test=# \d fbooking_user;
                                      Table "public.fbooking_user"
   Column    |          Type          | Collation | Nullable |                  Default                  
-------------+------------------------+-----------+----------+-------------------------------------------
 id          | integer                |           | not null | nextval('fbooking_user_id_seq'::regclass)
 username    | character varying(15)  |           | not null | 
 first_name  | character varying(35)  |           | not null | 
 last_name   | character varying(35)  |           | not null | 
 password    | character varying(64)  |           | not null | 
 state       | integer                |           | not null | 
 description | character varying(255) |           |          | 
Indexes:
    "fbooking_user_pkey" PRIMARY KEY, btree (id)
Referenced by:
    TABLE "fbooking_booking" CONSTRAINT "fbooking_booking_user_id_16171751_fk_fbooking_user_id" FOREIGN KEY (user_id) REFERENCES fbooking_user(id) DEFERRABLE INITIALLY DEFERRED
    TABLE "fbooking_guest" CONSTRAINT "fbooking_guest_user_id_25f1033b_fk_fbooking_user_id" FOREIGN KEY (user_id) REFERENCES fbooking_user(id) DEFERRABLE INITIALLY DEFERRED
    TABLE "fbooking_permissiontouser" CONSTRAINT "fbooking_permissiontouser_user_id_7ffdcbdf_fk_fbooking_user_id" FOREIGN KEY (user_id) REFERENCES fbooking_user(id) DEFERRABLE INITIALLY DEFERRED
`
No errors, no warnings, no duplicate fields

**Comment by Bright-Head** [comment]
Hi @venmer, I noticed that some checks failed with error
`TypeError: Unicode-objects must be encoded before hashing`
I'm not quite sure why, but I believe it makes sense to merge this branch with F-23 or anyway manage it. Also I left several comments, please, have a look.

**Comment by venmer** [comment]
@Bright-Head, thanks. CI checks have been passed.
I can't see any additional comments from you (maybe I just miss something here). Could you please point me?

File: flotski/fbooking/models.py [hunk 7]
```diff
@@ -4,50 +4,58 @@
...
     description = models.CharField(max_length=255, null=True)
 
 class Booking(models.Model):
     start_date = models.DateField()
     end_date = models.DateField()
-    user_id = models.ForeignKey(User, on_delete=models.CASCADE)
-    room_id = models.ForeignKey(Room, on_delete=models.CASCADE)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    room = models.ForeignKey(Room, on_delete=models.CASCADE)
     creation_date = models.DateField(auto_now_add=True)
```
[the remark this answers is not in the document]
Comment (venmer) [reply flotski/fbooking/models.py]: yes, this date will be added automatically with each new booking. and yes, it's just a date, as I remember we come up with a conclusion that we don't need exact time.

File: flotski/fbooking/models.py [hunk 8]
```diff
@@ -4,50 +4,58 @@
...
+    USER_STATE = (
+        (ACTIVE, 'active'),
+        (INACTIVE, 'inactive'),
+    )
+    description = models.CharField(max_length=255, null=True)
     username = models.CharField(max_length=15)
     first_name = models.CharField(max_length=35)
     last_name = models.CharField(max_length=35)
     password = models.CharField(max_length=64)
+    state = models.IntegerField(choices=USER_STATE,default=ACTIVE)
```
[the remark this answers is not in the document]
Comment (venmer) [reply flotski/fbooking/models.py]: ok

File: flotski/fbooking/models.py [hunk 9]
```diff
@@ -4,50 +4,58 @@
...
 
 class User(models.Model):
+    ACTIVE = 1
+    INACTIVE = 0
+    USER_STATE = (
+        (ACTIVE, 'active'),
+        (INACTIVE, 'inactive'),
+    )
+    description = models.CharField(max_length=255, null=True)
     username = models.CharField(max_length=15)
```
[the remark this answers is not in the document]
Comment (venmer) [reply flotski/fbooking/models.py]: yes, thanks

File: flotski/fbooking/models.py [hunk 10]
```diff
@@ -4,50 +4,58 @@
...
-        (MAINTENANCE, 'Maintenance')
+    ACTIVE = 1
+    INACTIVE = 0
+    ROOM_STATE = (
+        (ACTIVE, 'active'),
+        (INACTIVE, 'inactive'),
     )
     description = models.CharField(max_length=255, null=True)
-    state = models.CharField(max_length=1, choices=ROOM_STATE_CHOICES, default=VACANT)
     beds = models.IntegerField()
```
[the remark this answers is not in the document]
Comment (venmer) [reply flotski/fbooking/models.py]: ok

**Comment by Bright-Head** [comment]
Looks good, merged to master.

**Comment by venmer** [comment]
@Bright-Head, @sapopilas, PR was reopened, merge conflicts were resolved.
please approve

**Pull request merged**